jetty-proxy/src/test/java/org/eclipse/jetty/proxy/AsyncMiddleManServletTest.java

```diff
@@ -508,46 +508,52 @@ protected ContentTransformer newClientRequestContentTransformer(HttpServletReque
                     throw new NullPointerException("explicitly_thrown_by_test");
                 };
             }
-        });
+        };
+        startServer(new EchoHttpServlet());
+        startProxy(proxyServlet);
         startClient();
 
-        byte[] bytes = new byte[1024];
-        ContentResponse response = client.newRequest("localhost", serverConnector.getLocalPort())
-            .content(new BytesContentProvider(bytes))
-            .timeout(5, TimeUnit.SECONDS)
-            .send();
+        try (StacklessLogging ignored = new StacklessLogging(proxyServlet._log))
+        {
+            byte[] bytes = new byte[1024];
+            ContentResponse response = client.newRequest("localhost", serverConnector.getLocalPort())
+                .content(new BytesContentProvider(bytes))
+                .timeout(5, TimeUnit.SECONDS)
+                .send();
 
-        assertEquals(500, response.getStatus());
+            assertEquals(500, response.getStatus());
+        }
     }
 
     @Test
     public void testUpstreamTransformationThrowsAfterCommittingProxyRequest() throws Exception
     {
-        try (StacklessLogging ignored = new StacklessLogging(HttpChannel.class))
+        AsyncMiddleManServlet proxyServlet = new AsyncMiddleManServlet()
         {
-            startServer(new EchoHttpServlet());
-            startProxy(new AsyncMiddleManServlet()
+            @Override
+            protected ContentTransformer newClientRequestContentTransformer(HttpServletRequest clientRequest, Request proxyRequest)
             {
-                @Override
-                protected ContentTransformer newClientRequestContentTransformer(HttpServletRequest clientRequest, Request proxyRequest)
+                return new ContentTransformer()
                 {
-                    return new ContentTransformer()
-                    {
-                        private int count;
+                    private int count;
 
-                        @Override
-                        public void transform(ByteBuffer input, boolean finished, List<ByteBuffer> output)
-                        {
-                            if (++count < 2)
-                                output.add(input);
-                            else
-                                throw new NullPointerException("explicitly_thrown_by_test");
-                        }
-                    };
-                }
-            });
-            startClient();
+                    @Override
+                    public void transform(ByteBuffer input, boolean finished, List<ByteBuffer> output)
+                    {
+                        if (++count < 2)
+                            output.add(input);
+                        else
+                            throw new NullPointerException("explicitly_thrown_by_test");
+                    }
+                };
+            }
+        };
+        startServer(new EchoHttpServlet());
+        startProxy(proxyServlet);
```

Comment:
Restore inlining of `AsyncMiddleManServlet`.

Reply:
Restored inlining.